middleware/index.ts

```diff
@@ -282,6 +44,64 @@ function errorMessage(error: Error): any {
         .build();
 }
 
+function isAsyncRequest(type: string) {
+    return type && type.startsWith(ASYNC_CE_TYPE);
+}
+
+// Invoke given function again to fulfill async request; response is not handled
+async function invokeAsyncFn(logger: Logger, payload: any, headers: any): Promise<void> {
+    // host header is required
+    const hostKey = Object.keys(headers).find(k => X_FORWARDED_HOST === k.toLowerCase());
+    const hostUrl = headers[hostKey];
+    if (!hostUrl) {
+        throw new Error('Unable to determine host for async invocation');
+    }
+
+    const hostParts = hostUrl.split(':');
+    const protocolKey = Object.keys(headers).find(k => X_FORWARDED_PROTO === k.toLowerCase());
+    let isHttps = true;
+    if (headers[protocolKey]) {
+        isHttps = headers[protocolKey].toLowerCase() === 'https';
+    }
+
+    const options = {
+        hostname: hostParts[0],
+        port: hostParts[1] ? parseInt(hostParts[1]) : (isHttps ? 443 : 80),
+        method: 'POST',
+        path: '/',
+        headers: {...headers, [ASYNC_FULFILL_HEADER]: true}
+    };
+
+    // Invoke function and ignore response
+    const lib = isHttps ? https : http;
+    return new Promise((resolve, reject) => {
+        const expectedErrCode = 'EXPECTED_ECONNRESET';
+        const req = lib.request(options);
+        req.on('error', async (err) => {
+            if (req.destroyed && err['code'] === expectedErrCode) {
+                // Expected to terminate response
+                resolve();
+            } else {
+                // Return error for async request client
+                reject(err);
```

Comment:
Return errors encountered during the request back to original client to decide what to do.